The Genome Instability Pipeline (GIP) will solve your problem of analyzing whole genome sequencing (WGS) data making bioinformatics screenings faster and easier.

GIP allows the batch processing of multiple WGS experiments, including the mapping of short reads, the quantification of chromosomes, genes and genomic bins.

The results of GIP are summarized in a report page providing sequencing experiment and mapping statistics, graphical representations of genomic features quantifications, and excel tables.

GIP also enables giptools, a tool-suite allowing to Detect, Compare and Visualize

* Aneuploidy changes
* Gene copy number variants (CNVs)
* Single nucleotide variants (SNVs)
* Structural variants (SVs)


Documentation:
https://gip.readthedocs.io/en/latest/

pre-print:
bioRxiv 2021.06.15.448580; doi: https://doi.org/10.1101/2021.06.15.448580 

Citation:
If you are using GIP results in your upcoming publication(s), please always cite:
Sp√§th, Gerald F., and Giovanni Bussotti. "GIP: an open-source computational pipeline for mapping genomic instability from protists to cancer cells." Nucleic Acids Research 50.6 (2022): e36-e36


